src/components/userSign/signUp/SignUpForm.tsx

```diff
@@ -76,240 +48,35 @@ const SignUpForm = () => {
-              <InputRightElement>
-                {watchNickname && (
-                  <MdCancel cursor='pointer' onClick={() => resetField('nickname')} />
-                )}
-              </InputRightElement>
-            </InputGroup>
-            <Button
-              size='sm'
-              onClick={handleCheckNickname}
-              colorScheme={checkNickname ? 'green' : 'red'}>
-              중복 확인
-            </Button>
-          </Flex>
-          <Box height='4'>
-            {checkNickname && (
-              <Text pt='2' pl='2' fontSize='sm' color='green'>
-                사용 가능한 닉네임입니다.
-              </Text>
-            )}
-            <FormErrorMessage pl={2} fontSize='sm' color='red' pt='2'>
-              {errors.nickname?.message}
-            </FormErrorMessage>
-          </Box>
-        </FormControl>
-
-        <FormControl isInvalid={!!errors.password}>
-          <FormLabel fontSize='xs' fontWeight='bold' color='gray'>
-            비밀번호
-          </FormLabel>
-          <InputGroup size='md'>
-            <Input
-              id='password'
-              placeholder='비밀번호'
-              maxLength={12}
-              type='password'
-              {...register('password')}
-            />
-            <InputRightElement>
-              {watchPassword && (
-                <MdCancel cursor='pointer' onClick={() => resetField('password')} />
-              )}
-            </InputRightElement>
-          </InputGroup>
-          <Box height={4}>
-            <FormErrorMessage pl={2} fontSize='sm' color='red' pt='2'>
-              {errors.password?.message}
-            </FormErrorMessage>
-          </Box>
-        </FormControl>
-
-        <FormControl isInvalid={!!errors.passwordConfirm}>
-          <FormLabel fontSize='xs' fontWeight='bold' color='gray'>
-            비밀번호 확인
-          </FormLabel>
-          <InputGroup size='md'>
-            <Input
-              id='passwordConfirm'
-              placeholder='비밀번호 확인'
-              maxLength={12}
-              type='password'
-              {...register('passwordConfirm')}
-            />
-            <InputRightElement>
-              {watchPasswordConfirm && (
-                <MdCancel
-                  cursor='pointer'
-                  onClick={() => resetField('passwordConfirm')}
-                />
-              )}
-            </InputRightElement>
-          </InputGroup>
-          <Box height={4}>
-            <FormErrorMessage pl={2} fontSize='sm' color='red' pt={2}>
-              {errors.passwordConfirm?.message}
-            </FormErrorMessage>
-          </Box>
-        </FormControl>
-
+        <EmailInput
+          name='email'
+          control={control}
+          resetField={resetField}
+          trigger={trigger}
+          setError={setError}
+          checkEmail={checkEmail}
+          setCheckEmail={setCheckEmail}
+          certifyEmail={certifyEmail}
+          setCertifyEmail={setCertifyEmail}
```

Comment:
p5;

궁금한 점에 적어주신 것에 제 개인적인 의견을 한번 드려보자면....
useState의 get과 set을 한번에 넘겨주면 그나마 깔끔해질것같아요! 지금은 `certifyEmail`, `setCertifyEmail` 이렇게 따로따로 넘겨주고 있는데 이걸 묶어서
```suggestion
          certifyEmail={[certifyEmail, setCertifyEmail]}
```
이렇게 useState의 배열을 바로 넘겨주면 props 받는 쪽에서
```javascript
const [certifyEmail, setCertifyEmail] = props.certifyEmail
```
이런식으로 받고 사용하면 어떨까 생각드네요! 근데 타입지정을 어떻게 해줘야하는지랑.. 다른 문제가 또 생길수도있으니 그냥 참고용입니다^^